### PR TITLE
Create WebSockets.md

### PR DESCRIPTION
**Pull request by AKSHITHA-CHILUKA**
## Related Issue

[Cite any related issue(s) this pull request addresses. If none, simply state "None”]

## Description

[Please include a brief description of the changes or features added]

## Type of PR

- [ ] Bug fix
- [ ] Feature enhancement
- [x] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have followed the code style guidelines of this project.
- [x] I have checked for any existing open issues that my pull request may address.
- [x] I have ensured that my changes do not break any existing functionality.
- [x] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [x] I have read the resources for guidance listed below.
- [x] I have followed security best practices in my code changes.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Please read the following resources before submitting your contribution:

- [x] [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [x] [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [x] [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [x] [Markdown Guide](https://www.markdownguide.org/)
- [x] [MDX Documentation](https://mdxjs.com/docs/)
- [x] [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- [x] [Tabs: Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [x] [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [x] [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex) 

## Contribution Guidelines

Thank you for considering contributing to our project! To ensure smooth collaboration and effective contribution management, please adhere to the following guidelines:

### Issue Creation

1. **Limit on Issues:**
   - Each contributor is allowed to create a maximum of **4 issues per day**. This helps us manage and address issues efficiently.

### Contribution Levels

2. **Basic Contributions:**
   - This project is primarily focused on documentation. Most of the setup has been completed, so contributors will generally need to work on basic code tasks, such as writing tests.
   - For these tasks, issues will be assigned the **level1** label.

3. **Acknowledging Hard Work:**
   - If a contributor puts in significant effort on a task, the issue will be upgraded to **level2**. This is our way of recognizing and appreciating extra effort.

4. **Feature Additions and Component Work:**
   - Contributors working on new features or components using JSX/TSX will be assigned a level based on the complexity and quality of their work.
   - The more complex and valuable the contribution, the higher the level assigned.

### Level Definitions

- **Level1:**
  - Tasks are straightforward, such as fixing minor bugs, writing tests, or making simple documentation updates.
- **Level2:**
  - Tasks require more effort, such as addressing complex bugs, improving existing features, or making substantial documentation improvements.
- **Level3:**
  - Tasks are highly complex and involve significant new feature development, major refactoring, or extensive contributions to the project’s core components.

We look forward to your contributions and appreciate your effort in helping us improve the project!

### CODE REVIEW

**Comment by AKSHITHA-CHILUKA** [comment]
will the score be updated onto the leaderboard?

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
@AKSHITHA-CHILUKA Your code seems incomplete. Please provide the entire code and make sure it follows our guidelines.

**Review by AKSHITHA-CHILUKA — COMMENTED** [review]
completed the pending code

**Comment by AKSHITHA-CHILUKA** [comment]
Also sir can you update my score in the leaderboard

File: docs/WebSockets.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+// Handle WebSocket server errors
+wss.on('error', (error) => {
+  console.error(`WebSocket server error: ${error}`);
+});
+
+
+Client-Side Example:
+To interact with this WebSocket server from a client, you would typically use JavaScript in a web browser:
+
```
Comment (Ajay-Dhangar) [inline docs/WebSockets.md:71]: replace 

```md
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>WebSocket Client</title>
  <script>
    const ws = new WebSocket('ws://localhost:8080');

    // WebSocket open event
    ws.addEventListener('open', () => {
      console.log('Connected to WebSocket server');
    });

    // WebSocket message event
    ws.addEventListener('message', (event) => {
      console.log(`Received message from server: ${event.data}`);
    });

    // WebSocket close event
    ws.addEventListener('close', () => {
      console.log('Disconnected from WebSocket server');
    });

    // Function to send a message to the WebSocket server
    function sendMessage() {
      const message = document.getElementById('message').value;
      ws.send(message);
    }
  </script>
</head>
<body>
  <h1>WebSocket Client</h1>
  <input type="text" id="message" placeholder="Enter message">
  <button onclick="sendMessage()">Send</button>
</body>
</html>
```

to


````md
```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>WebSocket Client</title>
  <script>
    const ws = new WebSocket('ws://localhost:8080');

    // WebSocket open event
    ws.addEventListener('open', () => {
      console.log('Connected to WebSocket server');
    });

    // WebSocket message event
    ws.addEventListener('message', (event) => {
      console.log(`Received message from server: ${event.data}`);
    });

    // WebSocket close event
    ws.addEventListener('close', () => {
      console.log('Disconnected from WebSocket server');
    });

    // Function to send a message to the WebSocket server
    function sendMessage() {
      const message = document.getElementById('message').value;
      ws.send(message);
    }
  </script>
</head>
<body>
  <h1>WebSocket Client</h1>
  <input type="text" id="message" placeholder="Enter message">
  <button onclick="sendMessage()">Send</button>
</body>
</html>
```
````

File: docs/WebSockets.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  ws.on('error', (error) => {
+    console.error(`WebSocket error: ${error}`);
+  });
+});
+
+// Handle WebSocket server errors
+wss.on('error', (error) => {
+  console.error(`WebSocket server error: ${error}`);
+});
+
```
Comment (Ajay-Dhangar) [inline docs/WebSockets.md:67]: add <code>```</code>

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

**Comment by Ajay-Dhangar** [comment]
> Also sir can you update my score in the leaderboard

I don't do this but your score is update based on your work like if your PRs successful merge then calculate your levels and added your score in leaderboard  So, don't worry.

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**